### PR TITLE
add unit test coverage for pkg/kubelet/types/

### PR DESCRIPTION
**Pull request by zhoumingcheng**
Signed-off-by: zhoumingcheng <zhoumingcheng@beyondcent.com>

#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
Improve test coverage

Run test coverage command:
```
go test -cover ./pkg/kubelet/types/
```
Befor:
```
ok      k8s.io/kubernetes/pkg/kubelet/types      0.032s  coverage: 86.6% of statements
```
After:
```
ok      k8s.io/kubernetes/pkg/kubelet/types      0.019s  coverage: 100.0% of statements
```
#### Which issue(s) this PR fixes:

Fixes #

#### Special notes for your reviewer:
Thank you！
#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by zhoumingcheng** [comment]
/assign @yujuhong
please take a look, thanks!

**Comment by zhoumingcheng** [comment]
/assign @mrunalp
please take a look, thanks!

File: pkg/kubelet/types/pod_update_test.go [hunk 1]
```diff
@@ -356,3 +357,53 @@ func TestIsCriticalPodBasedOnPriority(t *testing.T) {
...
+		{
+			name:     "not critical if not static, mirror or without a priority",
+			pod:      getTestPod(nil, nil, ""),
+			expected: false,
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			isNodeCriticalPod := IsNodeCriticalPod(test.pod)
+			assert.Equal(t, test.expected, isNodeCriticalPod)
```
Comment (endocrimes) [inline pkg/kubelet/types/pod_update_test.go:406]:
```suggestion
			require.Equal(t, test.expected, isNodeCriticalPod)
```

(assert has the downside of not terminating the test, so errors stack and become hard to read/triage)

**Review by endocrimes — COMMENTED** [review]
/triage accepted
/priority backlog 

first pass looks mostly ok, will come back with a deeper look at `TestSortStatusesOfInitContainers` later

File: pkg/kubelet/types/types_test.go [hunk 2]
```diff
@@ -136,3 +137,48 @@ func TestSortInitContainerStatuses(t *testing.T) {
...
+		{
+			containers:     []v1.Container{{Name: "first"}, {Name: "second"}, {Name: "third"}, {Name: "fourth"}},
+			expectStatuses: []v1.ContainerStatus{{Name: "first"}, {Name: "third"}, {Name: "fourth"}},
+			statusMap:      map[string]*v1.ContainerStatus{"first": {Name: "first"}, "third": {Name: "third"}, "fourth": {Name: "fourth"}},
+		},
+	}
+	for _, data := range tests {
+		pod.Spec.InitContainers = data.containers
+		result := SortStatusesOfInitContainers(&pod, data.statusMap)
+		assert.Equal(t, result, data.expectStatuses, "Unexpected result from SortStatusesOfInitContainers: %v", result)
```
[the remark this answers is not in the document]
Comment (zhoumingcheng) [reply pkg/kubelet/types/types_test.go]: Thanks for your suggestion, I have revised the above question.

File: pkg/kubelet/types/types_test.go [hunk 3]
```diff
@@ -136,3 +138,48 @@ func TestSortInitContainerStatuses(t *testing.T) {
...
+	}
+}
+
+func TestGet(t *testing.T) {
+	timestamp := NewTimestamp()
+	timestampString := timestamp.GetString()
+	timeParse, _ := time.Parse(RFC3339NanoLenient, timestampString)
+	timeResult := ConvertToTimestamp(timestampString).Get()
+	assert.Equal(t, timeResult, timeParse)
+}
```
Comment (s-kawamura-w664) [inline pkg/kubelet/types/types_test.go:185]:
```suggestion
}
func TestNewTimestamp(t *testing.T) {
	timeStart := time.Now()
	timestamp := NewTimestamp()
	timeEnd := time.Now()
	assert.WithinDuration(t, timestamp.Get(), timeStart, timeEnd.Sub(timeStart))
}
```
I think you needed to use NewTimestamp() for 100% of coverage.
I think your test code alone would be fine, but
I would suggest a test code like the above for NewTimestamp(),

Comment (s-kawamura-w664) [reply pkg/kubelet/types/types_test.go:185]: My suggestion follows the comments in the source code below.

```
// NewTimestamp returns a Timestamp object using the current time.
func NewTimestamp() *Timestamp {
	return &Timestamp{time.Now()}
}
```

Comment (zhoumingcheng) [reply pkg/kubelet/types/types_test.go:184]: Thank you for your suggestion, which I also agree with and adopt.

**Comment by zhoumingcheng** [comment]
/assign @neolit123
please take a look, thanks!

**Review by neolit123 — COMMENTED** [review]
Lgtm on a quick look, but up to the kubelet maintainers for proper review.

**Comment by zhoumingcheng** [comment]
/assign @klueska
please take a look, thanks!

**Comment by s-kawamura-w664** [comment]
The update against my comment is LGTM. But I am not sig-node member.
I would like to wait for the signode members' reviews.

**Comment by zhoumingcheng** [comment]
/assign @sjenning
please take a look, thanks!

**Comment by zhoumingcheng** [comment]
/assign @dashpole
please take a look, thanks!

**Comment by derekwaynecarr** [comment]
Thank you for the test updates.

/lgtm
/approve

**Pull request merged**